.github/workflows/ci_nightly.yml

```diff
@@ -0,0 +1,440 @@
+      - uses: actions/upload-artifact@v4
+        with:
+          name: coverage-lcov-${{ matrix.type }}
+          path: coverage/lcov
+      - uses: actions/upload-artifact@v4
+        with:
+          name: coverage-html-${{ matrix.type }}
+          path: coverage/html
+
+  upload_coverage:
+    name: "Upload Coverage"
+    runs-on: ubuntu-latest
+    needs:
+      - cargo_nextest
+      - py_backward_compat
+      - py_db_migration
+      - py_sanity_checks
+      - py_genesis_check
+      - py_upgradability
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: coverage-codecov-*
+          merge-multiple: true
+      # Keep the number of uploads here in sync with codecov.yml’s after_n_build value
+      # codecov will send a comment only after having receidev this number of uploads.
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          files: unit-linux.json
+          fail_ci_if_error: true
+          flags: unittests,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          files: unit-linux-nightly.json
+          fail_ci_if_error: true
+          flags: unittests,linux-nightly
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          files: unit-macos.json
+          fail_ci_if_error: true
+          flags: unittests,macos
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          files: py-backward-compat.json
+          fail_ci_if_error: true
+          flags: pytests,backward-compatibility,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          files: py-db-migration.json
+          fail_ci_if_error: true
+          flags: pytests,db-migration,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          files: py-sanity-checks.json
+          fail_ci_if_error: true
+          flags: pytests,sanity-checks,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          files: py-genesis-check.json
+          fail_ci_if_error: true
+          flags: pytests,genesis-check,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          files: py-upgradability.json
+          fail_ci_if_error: true
+          flags: pytests,upgradability,linux
+
+  windows_public_libraries_check:
+    name: "Windows check for building public libraries"
+    runs-on: "windows-latest"
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+      - uses: taiki-e/install-action@9b5b983efc779f85e5e5d11539f005e85ccb27ff
+        with:
+          tool: just
+      - run: just check_build_public_libraries
```

Comment:
All of these steps can be removed as they won't work.

Reply:
can't test it properly until in master and GHA recognizes new workflow and allows me to run it